ord_schema/units.py

```diff
@@ -146,27 +160,49 @@ def __init__(self,
                     self._resolver[string_unit] = (message, unit)
         # Values must have zero or one decimal point. Whitespace between the
         # value and the unit is optional.
-        self._pattern = re.compile(r'(\d+.?\d*)\s*(\w+)')
+        self._pattern = re.compile(
+            r'(-?\d+\.?\d*(?:[eE]-?\d+)?)(?:[-±](-?\d+\.?\d*))?\s*'
+            r'([\w\sμ°]+)\.?')
 
-    def resolve(self, string: str) -> ord_schema.UnitMessage:
+    def resolve(self,
+                string: str,
+                allow_range: bool = False) -> ord_schema.UnitMessage:
         """Resolves a string into a message containing a value with units.
 
         Args:
             string: The string to parse; must contain a numeric value and a
                 string unit. For example: "1.25 h".
+            allow_range: If True, ranges like "1-2 h" can be provided and the
+                average value will be reported along with the standard
+                deviation.
 
         Returns:
             Message containing a numeric value with units listed in the schema.
 
         Raises:
-            ValueError: If string does not contain a value with units.
+            ValueError: if string does not contain a value with units, or if
+                the value is invalid.
         """
         # NOTE(kearnes): Use fullmatch() to catch cases with multiple matches.
-        match = self._pattern.fullmatch(string.strip())
+        match = self._pattern.fullmatch(string.strip().replace('−', '-'))
         if not match:
             raise ValueError(
                 f'string does not contain a value with units: {string}')
-        value, string_unit = match.groups()
+        value, range_value, string_unit = match.groups()
+        precision = None
+        if range_value is not None:
+            if '±' in string:
+                value = float(value)
+                precision = float(range_value)
+            elif not allow_range:
+                raise ValueError('string appears to contain a range of values '
+                                 f'but allow_range is False: {string}')
+            else:
+                values = np.asarray([value, range_value], dtype=float)
+                value = values.mean()
+                precision = values.std()
```

Comment:
Should we use the standard devision here or `max - min`? I think the only difference is a factor of $\sqrt{2}$, but my inclination is that a value of "70-80% yield" should be transcribed as $70 \pm 5$.

Reply:
Just to clarify:

* Do you mean `75 \pm 5`?
* `np.std([70, 80]) == 5.0`; so it might do what you want already?

Reply:
It would look that way. I guess `ddof=0` by default so never mind!